src/main/scala/is/hail/expr/Type.scala

```diff
@@ -190,6 +187,10 @@ sealed abstract class Type extends Serializable { self =>
   def byteSize: Int = throw new NotImplementedError(toString)
 
   def alignment: Int = byteSize
+
+  /*  Fundamental types are types that can be handled natively by RegionValueBuilder: primitive
+      types, Array and Struct. */
+  def fundamentalType: Type = this
```

Comment:
I think this is the right thing - each type contains its representation in primitives. 

I don't mean remove TVariant and TAltAllele, I mean remove the ComplexType class, since `def representation`is just `fundamentalType` for types that actually have a different fundamental type than `this`